TryOn/Sources/TryOnViewController.swift

```diff
@@ -0,0 +1,373 @@
+open class TryOnViewController: UIViewController {
+    open private(set) lazy var tryOnView: WsneakersUISDKView = {
+        let view = WsneakersUISDKView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.supportedOrientationMask = .all
+
+        return view
+    }()
+
+    var wsneakersSession: WannaSDKSession?
+
+    @IBOutlet weak var videoRecordButton: UIButton!
+    @IBOutlet weak var activity: UIActivityIndicatorView!
+    @IBOutlet weak var progress: UIProgressView!
+    @IBOutlet weak var modelId: UILabel!
+
+    private final var storage: WsneakersUISDKRenderModelStorage?
+    private final var renderModels: [String] = []
+
+    private var currentTask: WsneakersUISDKDownloadTask?
+    private var currentIndex = 0
+    private var videoRecordRunning = false
+
+    deinit {
+        wsneakersSession?.stop { _ in
+        }
+    }
+
+    // MARK: - Override
+
+    open override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupTryOnView()
+        progress.isHidden = true
+
+        loadModel(with: currentIndex)
+    }
+
+    open override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        // Start showing the virtual try-on
+        if let wsneakersSession, !wsneakersSession.isDrawing() {
+            wsneakersSession.startDrawing()
+        }
+    }
+
+    open override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        // Stop rendering the virtual try-on
+        wsneakersSession?.stopDrawing()
+    }
+}
+
+// MARK: - Dependency injection
+
+@available(iOS 13.0, *)
+public extension TryOnViewController {
+    // Set the view to use the previously created session and storage objects,
+    // and work with the list of 3D models with one already selected.
+    func set(
+        session: WannaSDKSession,
+        storage: WsneakersUISDKRenderModelStorage,
+        renderModels: [String],
+        selected: Int
+    ) {
+        wsneakersSession = session
+        currentIndex = selected
+
+        self.storage = storage
+        self.renderModels = renderModels
+
+        session.start()
+    }
+}
+
+// MARK: - Model selection
+
+@MainActor
+@available(iOS 13.0, *)
+private extension TryOnViewController {
+    @IBAction func onNextRenderModel(_ sender: Any) {
+        var nextIndex = currentIndex + 1
+        if (nextIndex == renderModels.count) {
+            nextIndex = 0
+        }
+
+        loadModel(with: nextIndex)
+    }
+
+    @IBAction func onPreviousRenderModel(_ sender: Any) {
+        var nextIndex = currentIndex - 1
+        if (nextIndex < 0) {
+            nextIndex = renderModels.count - 1
+        }
+
+        loadModel(with: nextIndex)
+    }
+}
+
+@available(iOS 13.0, *)
+private extension TryOnViewController {
+    func setRenderModel(_ renderModel: WsneakersUISDKRenderModel, index: Int) {
+        wsneakersSession?.change(renderModel) { [weak self] error in
+            Task {
```

Comment:
Is current Actor `MainActor`? Shouldn't we define it explicitly?

Reply:
yes, it's `MainActor` because `onModelLoadFailed(error:)` is `@MainActor` and compiler automatically dispatches it on main thread.
it's also automatically dispatching to `.userInitiated` queue when code uses `actor`'s func

```swift
func newConcurrencyCode() {
    Task {
        // Executes asynchronously on the main thread
        // Depending on whether the newConcurrencyCode function
        // is implemented by an object protected by MainActor previously
    }
    
    Task { @MainActor in
        // Executes asynchronously on the main thread
        // explicitly
    }
    
    Task(priority: .background) {
        // Executes asynchronously on any thread
        // with background priority
    }
}
```
https://onnerb.medium.com/swift-concurrency-fd42c072234e